schema/ksl-schema.kdl

```diff
@@ -0,0 +1,925 @@
+                    type string
+                    format url irl
+                }
+                // TODO: expand a bit more about what it means for the enum to be `allow-others`.
+                prop rel about="Link relationship\n\nThe relation between the current entity and the URL." {
+                    required
+                    type string
+                    enum self documentation {
+                        allow-others
+                    }
+                }
+                prop lang about="Link language\n\nLanguage of the destination document." {
+                    ref metadata-lang
+                }
+            }
+            node license about="""
+                Schema license
+
+                The license(s) that the schema is licensed under.
+                """ {
+                repeatable
+                arg description="Name of the used license" {
+                    type string
+                }
+                prop spdx description="An SPDX license identifier" {
+                    type string
+                    // TODO: validation?
+                }
+                prop path about="Path to a local license file" {
+                    type string
+                }
+                prop url about="URL to an externally-stored license" {
+                    type string
+                    format url url-reference irl irl-reference
+                }
+                children {
+                    node link {
+                        ref metadata-link
+                    }
+                }
+            }
+            node published about="""
+                Schema publication date
+
+                Date or data+time when the schema was published.
+                """ {
+                arg about="Publication date" {
+                    type string
+                    format date date-time
+                }
+            }
+            node modified about="""
+                Schema modification date
+
+                When the schema was modified. If used multiple times, the most
+                recent date will be considered 'latest'.
+                """ {
+                repeatable
+                args about="Modification date" {
+                    type string
+                    format date date-time
+                }
+            }
+            node version about="""
+                Schema semver version
+
+                The version number of this version of the schema, in semver
+                format.
+                """ {
+                arg about="Semver version number" {
+                    type string
+                    // https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string.
+                    pattern #"^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"#
```

Comment:
Is this worth adding to `format` / `validate-as` and KDLs reserved type annotations? https://github.com/kdl-org/kdl/blob/main/draft-marchan-kdl2.md#reserved-type-annotations-for-strings

Personally, I can imagine more KDL documents with Semver strings than some of the other types already reserved! If you're already wanting to add something like `kpath`, then I reckon this would be helpful and would save other people the research and copy-paste of that regex beast!

Reply:
I was thinking this, yes